apollo-federation/src/query_plan/query_planner.rs

```diff
@@ -995,10 +995,7 @@ type User
         "###);
     }
 
-    // TODO: This fails with "Subgraph unexpectedly does not use federation spec"
-    // which seems...unusual
     #[test]
-    #[ignore]
```

Comment:
since the query plan is different should this be actually snapshot assertion vs updating the test?

Reply:
I think the old snapshot was generated from Rust version. I re-generated the new snapshot using the JS federation. In other words, I did not accept it from `cargo insta review`.

Reply:
Here's the test I used in `query-planner-js/src/__tests__/supergraphBackwardCompatibility.test.ts`:
```ts
        test('plan_simple_root_field_query_for_multiple_subgraphs', () => {
          qp = new QueryPlanner(supergraph);

          const operation = operationFromDocument(
            api,
            gql`
              {
                userById(id: 1) {
                  name
                  email
                }
                bestRatedProducts {
                  id
                  avg_rating
                }
              }
            `,
          );

          const plan = qp.buildQueryPlan(operation);
          expect(plan).toMatchInlineSnapshot(`
            QueryPlan {
              Parallel {
                Fetch(service: "accounts") {
                  {
                    userById(id: 1) {
                      name
                      email
                    }
                  }
                },
                Sequence {
                  Fetch(service: "reviews") {
                    {
                      bestRatedProducts {
                        __typename
                        id
                        ... on Book {
                          __typename
                          id
                          reviews {
                            rating
                          }
                        }
                        ... on Movie {
                          __typename
                          id
                          reviews {
                            rating
                          }
                        }
                      }
                    }
                  },
                  Flatten(path: "bestRatedProducts.@") {
                    Fetch(service: "products") {
                      {
                        ... on Book {
                          __typename
                          id
                          reviews {
                            rating
                          }
                        }
                        ... on Movie {
                          __typename
                          id
                          reviews {
                            rating
                          }
                        }
                      } =>
                      {
                        ... on Book {
                          avg_rating
                        }
                        ... on Movie {
                          avg_rating
                        }
                      }
                    },
                  },
                },
              },
            }
          `);
        });
```